integration/integration_test.go

```diff
@@ -0,0 +1,454 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+)
+
+var integration = flag.Bool("integration", false, "run integration tests")
+
+// image such as docker.io/kinvolk/gadget:latest
+var image = flag.String("image", "", "gadget container image")
+
+var githubCI = flag.Bool("github-ci", false, "skip some tests which cannot be run on GitHub CI due to host kernel not BPF ready")
+
+func runCommands(cmds []*command, t *testing.T) {
+	for _, cmd := range cmds {
+		cmd.run(t)
+	}
```

Comment:
I think you need to modify this function so use defer, otherwise the cleanup operations won't be executed if there's an error:

https://github.com/kinvolk/inspektor-gadget/blob/d0154900b44c0fc89a1b10b8dd206e12f31aeb71/integration/integration_test.go#L31-L54